fido/src/main/java/com/yubico/yubikit/fido/ctap/FingerprintBioEnrollment.java

```diff
@@ -0,0 +1,496 @@
+
+    public FingerprintBioEnrollment(
+            Ctap2Session ctap,
+            PinUvAuthProtocol pinUvAuthProtocol,
+            byte[] pinUvToken) throws IOException, CommandException {
+        super(ctap, BioEnrollment.MODALITY_FINGERPRINT);
+        this.pinUvAuth = pinUvAuthProtocol;
+        this.pinUvToken = pinUvToken;
+        this.cachedSensorInfo = readFingerprintSensorInfo(ctap);
+    }
+
+    private Map<Integer, ?> call(
+            Integer subCommand,
+            @Nullable Map<?, ?> subCommandParams,
+            @Nullable CommandState state) throws IOException, CommandException {
+        return call(subCommand, subCommandParams, state, true);
+    }
+
+    private Map<Integer, ?> call(
+            Integer subCommand,
+            @Nullable Map<?, ?> subCommandParams,
+            @Nullable CommandState state,
+            boolean authenticate) throws IOException, CommandException {
+        byte[] pinUvAuthParam = null;
+        if (authenticate) {
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            output.write(MODALITY_FINGERPRINT);
+            output.write(subCommand);
+            if (subCommandParams != null) {
+                Cbor.encodeTo(output, subCommandParams);
+            }
+            pinUvAuthParam = pinUvAuth.authenticate(pinUvToken, output.toByteArray());
+        }
+
+        return ctap.bioEnrollment(
+                modality,
+                subCommand,
+                subCommandParams,
+                pinUvAuth.getVersion(),
+                pinUvAuthParam,
+                null,
+                state);
+    }
+
+    /**
+     * Get fingerprint sensor info.
+     *
+     * @param ctap CTAP2 session
+     * @return A dict containing FINGERPRINT_KIND, MAX_SAMPLES_REQUIRES and
+     * MAX_TEMPLATE_FRIENDLY_NAME.
+     * @throws IOException      A communication error in the transport layer.
+     * @throws CommandException A communication error in the protocol layer.
+     * @see <a href="https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#getFingerprintSensorInfo">Get fingerprint sensor info</a>
+     */
+    public static SensorInfo readFingerprintSensorInfo(Ctap2Session ctap)
+            throws IOException, CommandException {
+
+        final Map<Integer, ?> result = ctap.bioEnrollment(
+                MODALITY_FINGERPRINT,
+                CMD_GET_SENSOR_INFO,
+                null,
+                null,
+                null,
+                null,
+                null);
+
+        return new SensorInfo(
+                Objects.requireNonNull((Integer) result.get(RESULT_FINGERPRINT_KIND)),
+                Objects.requireNonNull((Integer) result.get(RESULT_MAX_SAMPLES_REQUIRED)),
+                Objects.requireNonNull((Integer) result.get(RESULT_MAX_TEMPLATE_FRIENDLY_NAME))
+        );
+    }
+
+    public SensorInfo getCachedSensorInfo() {
+        return cachedSensorInfo;
+    }
```

Comment:
Is there a need to have a cached version of this in the class, and to read it in the constructor?

Reply:
no reason, fixing